libs/application/templates/p-sign/src/fields/PhotoUpload/index.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react'
+import { Box, InputFileUpload, Stack } from '@island.is/island-ui/core'
+import { m } from '../../lib/messages'
+import { useLocale } from '@island.is/localization'
+import { Application, FieldBaseProps } from '@island.is/application/core'
+import { useFormContext } from 'react-hook-form'
+import Photo from '../Photo'
+
+interface PhotoUploadProps {
+  application: Application
+}
+
+const PhotoUpload = ({ application }: PhotoUploadProps & FieldBaseProps) => {
+  const { formatMessage } = useLocale()
+  const [image, setImage] = useState(
+    (application.answers.photoAttachment as string) ?? '',
+  )
+  const [fileName, setFileName] = useState(
+    application.answers.attachmentFileName
+      ? [
+          {
+            name: application.answers.attachmentFileName,
+            status: 'done',
+          } as any,
+        ]
+      : [],
+  )
+  const { setValue } = useFormContext()
+
+  const updateApp = async (file: string, name: string) => {
+    setValue('photoAttachment', file)
+    setValue('attachmentFileName', name)
+  }
+
+  const onChange = (newFiles: File[]) => {
+    setFileName([{ name: newFiles[0].name, status: 'done' }] as any)
+    const fileReader = new FileReader()
+
+    fileReader.onload = function (fileLoadedEvent) {
+      const srcData = fileLoadedEvent.target?.result //base64
+      setImage(srcData as string)
+      updateApp(srcData as string, newFiles[0].name)
```

Comment:
We need to use the application attachment functionality for this rather than saving uploads as a base64 strings in the attachment. These photos/files can easily be multiple megabytes which can put a strain on the "answers" JSON column in the application table. Especially when fetching and updating it multiple times. It breaks consistency between templates and might hurt later on.

You can find examples of how attachments are used and submitted in other templates.

The tricky thing is how to preview the photo/upload, especially after reloading the page. Maybe that's ok for v1?

Unfortunately, there isn't any way to get a public url for attachments yet. It is a bit tricky to implement with regard to security, but it would be worthwhile for this and future applications. The feature may take a week or more for your team. Please sync with your PO and me before moving forward with this.

Reply:
Yes, we've now also realised the downsides of this implementation. 
Currently looking for the possible ways of implementing the preview.